src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -160,6 +161,7 @@ public virtual async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsyn
                 .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First());
 
             IEnumerable<TransitivePackageReference> transitivePackagesWithOrigins;
+            TelemetryActivity.EmitTelemetryEvent(new TransitiveDependenciesCounterfactualEvent());
```

Comment:
I understand that there's no big concern about the volume, but this gets called a lot. Multiple times during solution load. Have you checked how many events would get fired if you say opened nuget.sln with the exp enabled?

Reply:
actually exp being enabled doesn't matter....but in general how many events get emitted because of that?

Reply:
My fear is that it might double the total `vs/nuget` events.

Reply:
Expecitng increase of 1.33% of total NuGet events.

Reply:
Added a lock to reduce number of events.

Reply:
My comment was about events fired after solutioin load for no good action.

Reply:
Added static mutexes to emit counterfactual events at least once, but not so often. Closing this for now.